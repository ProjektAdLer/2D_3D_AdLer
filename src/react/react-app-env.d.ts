/// <reference types="react-scripts" />

// Images
declare module '*.jpg';
declare module '*.png';
declare module '*.env';

// 3D types
declare module '*.glb';
declare module '*.stl';

declare module '*.jpeg';
declare module '*.gif';
declare module '*.mp4';
