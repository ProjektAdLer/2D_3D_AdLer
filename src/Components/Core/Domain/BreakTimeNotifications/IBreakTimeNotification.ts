export default interface IBreakTimeNotification {
  titleKey: string; // key reference string inside localize language package
  titleMessageKeys: string[]; // keys reference strings inside localize language package
  images: string[];
  seenBefore: boolean;
}
