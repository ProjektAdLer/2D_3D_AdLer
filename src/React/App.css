body {
  margin: 0;
  height: 100vh;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

div{
  width: 100%;
  height: 100%;
}

section{
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 120px;
  height: 100vh;
  background-color: #fff;
}

button{
  width: 100px;
  margin: 10px;
  z-index: 5;
  position: fixed;
  left: 0;
  top: 0;
}

canvas {
  width: calc(100% - 100px);
  height: 100%;
  margin-left: 100px;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal-content {
  width: 1000px;
  background-color: #fff;
}

.modal-header,
.modal-footer {
  padding: 10px;
  height: 60px;
}

.modal-title {
  margin: 0;
}

.modal-body {
  padding: 10px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}
