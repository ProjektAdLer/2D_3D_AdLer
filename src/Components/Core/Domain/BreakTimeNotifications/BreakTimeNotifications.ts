// Small Break alles au√üer H und S
// Small Break B3
import SmallBreakB3Slide1Image from "../../../../Assets/BreakTimeAssets/B3/ps-B3-slide1.png";
import SmallBreakB3Slide2Image from "../../../../Assets/BreakTimeAssets/B3/smallBreakContent-B3-slide2.png";
// Small Break B7
import SmallBreakB7Slide1Image from "../../../../Assets/BreakTimeAssets/B7/smallBreakContent-B7-slide1.png";
import SmallBreakB7Slide2Image from "../../../../Assets/BreakTimeAssets/B7/smallBreakContent-B7-slide2.png";
// Small Break B8
import SmallBreakB8Slide1Image from "../../../../Assets/BreakTimeAssets/B8/smallBreakContent-B8-slide1.png";
import SmallBreakB8Slide2Image from "../../../../Assets/BreakTimeAssets/B8/smallBreakContent-B8-slide2.png";
// Small Break E2
import SmallBreakE2Slide1Image from "../../../../Assets/BreakTimeAssets/E2/smallBreakContent-E2-slide1.png";
import SmallBreakE2Slide2Image from "../../../../Assets/BreakTimeAssets/E2/smallBreakContent-E2-slide2.png";
import SmallBreakE2Slide3Image from "../../../../Assets/BreakTimeAssets/E2/smallBreakContent-E2-slide3.png";
// Small Break E4
import SmallBreakE4Slide1Image from "../../../../Assets/BreakTimeAssets/E4/smallBreakContent-E4-slide1.png";
// Small Break E10
import SmallBreakE10Slide1Image from "../../../../Assets/BreakTimeAssets/E10/smallBreakContent-E10-slide1.png";
import SmallBreakE10Slide2Image from "../../../../Assets/BreakTimeAssets/E10/smallBreakContent-E10-slide2.png";
// Small Break L1
import SmallBreakL1Slide1Image from "../../../../Assets/BreakTimeAssets/L1/smallBreakContent-L1-slide1.png";
import SmallBreakL1Slide2Image from "../../../../Assets/BreakTimeAssets/L1/smallBreakContent-L1-slide2.png";
import SmallBreakL1Slide3Image from "../../../../Assets/BreakTimeAssets/L1/smallBreakContent-L1-slide3.png";
// Small Break L3
import SmallBreakL3Slide1Image from "../../../../Assets/BreakTimeAssets/L3/smallBreakContent-L3-slide1.png";
import SmallBreakL3Slide2Image from "../../../../Assets/BreakTimeAssets/L3/smallBreakContent-L3-slide2.png";
// Small Break L5
import SmallBreakL5Slide1Image from "../../../../Assets/BreakTimeAssets/L5/smallBreakContent-L5-slide1.png";
import SmallBreakL5Slide2Image from "../../../../Assets/BreakTimeAssets/L5/smallBreakContent-L5-slide2.png";
// Small Break U3
import SmallBreakU3Slide1Image from "../../../../Assets/BreakTimeAssets/U3/smallBreakContent-U3-slide1.png";
import SmallBreakU3Slide2Image from "../../../../Assets/BreakTimeAssets/U3/smallBreakContent-U3-slide2.png";
import SmallBreakU3Slide3Image from "../../../../Assets/BreakTimeAssets/U3/smallBreakContent-U3-slide3.png";
import SmallBreakU3Slide4Image from "../../../../Assets/BreakTimeAssets/U3/smallBreakContent-U3-slide4.png";
// Medium Break = S
// Medium Break S1
import MediumBreakS1Slide1Image from "../../../../Assets/BreakTimeAssets/S1/mediumBreakContent-S1-slide1.png";
import MediumBreakS1Slide2Image from "../../../../Assets/BreakTimeAssets/S1/mediumBreakContent-S1-slide2.png";
import MediumBreakS1Slide3Image from "../../../../Assets/BreakTimeAssets/S1/mediumBreakContent-S1-slide3.png";
import MediumBreakS1Slide4Image from "../../../../Assets/BreakTimeAssets/S1/mediumBreakContent-S1-slide4.png";
// Medium Break S3
import MediumBreakS3Slide1Image from "../../../../Assets/BreakTimeAssets/S3/mediumBreakContent-S3-slide1.png";
import MediumBreakS3Slide2Image from "../../../../Assets/BreakTimeAssets/S3/mediumBreakContent-S3-slide2.png";
import MediumBreakS3Slide3Image from "../../../../Assets/BreakTimeAssets/S3/mediumBreakContent-S3-slide3.png";
// Long Break = H
//Long Break H1
import LongBreakH1Slide1Image from "../../../../Assets/BreakTimeAssets/H1/longBreakContent-H1-slide1.png";
import LongBreakH1Slide2Image from "../../../../Assets/BreakTimeAssets/H1/longBreakContent-H1-slide2.png";
import LongBreakH1Slide3Image from "../../../../Assets/BreakTimeAssets/H1/longBreakContent-H1-slide3.png";

import IBreakTimeNotification from "./IBreakTimeNotification";

export const shortBreakTimeNotificationContents: IBreakTimeNotification[] = [
  {
    titleKey: "B3",
    titleMessageKeys: ["shortBreakBase", "infoMobilization"],
    images: [SmallBreakB3Slide1Image, SmallBreakB3Slide2Image],
    seenBefore: false,
  },
  {
    titleKey: "B7",
    titleMessageKeys: ["shortBreakBase", "infoMobilization"],
    images: [SmallBreakB7Slide1Image, SmallBreakB7Slide2Image],
    seenBefore: false,
  },
  {
    titleKey: "B8",
    titleMessageKeys: ["shortBreakBase"],
    images: [SmallBreakB8Slide1Image, SmallBreakB8Slide2Image],
    seenBefore: false,
  },
  {
    titleKey: "E2",
    titleMessageKeys: ["shortBreakBase", "infoNutrition"],
    images: [
      SmallBreakE2Slide1Image,
      SmallBreakE2Slide2Image,
      SmallBreakE2Slide3Image,
    ],
    seenBefore: false,
  },
  {
    titleKey: "E4",
    titleMessageKeys: ["shortBreakBase"],
    images: [SmallBreakE4Slide1Image],
    seenBefore: false,
  },
  {
    titleKey: "E10",
    titleMessageKeys: ["shortBreakBase", "infoNutrition"],
    images: [SmallBreakE10Slide1Image, SmallBreakE10Slide2Image],
    seenBefore: false,
  },
  {
    titleKey: "L1",
    titleMessageKeys: ["shortBreakBase"],
    images: [
      SmallBreakL1Slide1Image,
      SmallBreakL1Slide2Image,
      SmallBreakL1Slide3Image,
    ],
    seenBefore: false,
  },
  {
    titleKey: "L3",
    titleMessageKeys: ["shortBreakBase"],
    images: [SmallBreakL3Slide1Image, SmallBreakL3Slide2Image],
    seenBefore: false,
  },
  {
    titleKey: "L5",
    titleMessageKeys: ["shortBreakBase"],
    images: [SmallBreakL5Slide1Image, SmallBreakL5Slide2Image],
    seenBefore: false,
  },
  {
    titleKey: "U3",
    titleMessageKeys: ["shortBreakBase", "infoSeatingPosition"],
    images: [
      SmallBreakU3Slide1Image,
      SmallBreakU3Slide2Image,
      SmallBreakU3Slide3Image,
      SmallBreakU3Slide4Image,
    ],
    seenBefore: false,
  },
];

export const mediumBreakTimeNotificationContents: IBreakTimeNotification[] = [
  {
    titleKey: "S1",
    titleMessageKeys: ["mediumBreakBase", "infoMusic"],
    images: [
      MediumBreakS1Slide1Image,
      MediumBreakS1Slide2Image,
      MediumBreakS1Slide3Image,
      MediumBreakS1Slide4Image,
    ],
    seenBefore: false,
  },
  {
    titleKey: "S3",
    titleMessageKeys: ["mediumBreakBase", "infoWhy"],
    images: [
      MediumBreakS3Slide1Image,
      MediumBreakS3Slide2Image,
      MediumBreakS3Slide3Image,
    ],
    seenBefore: false,
  },
];

export const longBreakTimeNotificationContents: IBreakTimeNotification[] = [
  {
    titleKey: "H1",
    titleMessageKeys: ["longBreakBase", "infoStroll"],
    images: [
      LongBreakH1Slide1Image,
      LongBreakH1Slide2Image,
      LongBreakH1Slide3Image,
    ],
    seenBefore: false,
  },
];
