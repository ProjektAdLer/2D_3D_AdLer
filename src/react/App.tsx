import React from 'react';
import './App.css';
import BabylonCanvas from './components/BabylonCanvas';

function App() {
	return (
		<React.Fragment>
			<BabylonCanvas style={{ width: '100%', position: 'absolute' }} />
		</React.Fragment>
	);
}

export default App;
