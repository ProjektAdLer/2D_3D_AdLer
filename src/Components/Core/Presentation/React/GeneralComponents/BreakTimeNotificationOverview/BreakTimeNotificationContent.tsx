import { Style } from "@babylonjs/gui";
import { useState } from "react";
import { useTranslation } from "react-i18next";
import IBreakTimeNotification from "src/Components/Core/Domain/BreakTimeNotifications/IBreakTimeNotification";
import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";

export default function BreakTimeNotificationContent({
  breakTimeNotification,
}: {
  breakTimeNotification: IBreakTimeNotification;
}) {
  const [currentSlide, setCurrentSlide] = useState(0);

  const { t: translate } = useTranslation("breakTime");

  return (
    <div className="pb-4 max-h-[90vh] portrait:max-w-[90vw]">
      <div className="flex flex-col items-center gap-6 mx-0 my-auto slider-wrapper">
        <div className="flex flex-col gap-4 overflow-y-auto items-center justify-center slider portrait:max-w-[90vw] max-h-[90vh]">
          {/* Description Title Text */}
          <p className="w-full pb-4 pl-6 text-sm font-bold md:text-xl lg:text-xl max-w-[60vw] ultraWide:max-w-[40vw] text-adlerdarkblue">
            {breakTimeNotification.titleMessageKeys.map((key) => {
              return translate(key) + " ";
            })}
          </p>

          {/* Image Slider */}
          <div className="w-full flex flex-row justify-center align-center">
            {/* Button left*/}
            <div className="flex justify-start items-center w-12">
              {currentSlide > 0 && (
                <StyledButton
                  shape="smallSquare"
                  className=" bg-whitetrans rounded-xl w-full "
                  onClick={() => {
                    setCurrentSlide(currentSlide - 1);
                  }}
                >
                  <p className="rotate-180">&#10140;</p>
                </StyledButton>
              )}
            </div>
            {/* Image Slide */}
            <img
              id="slide-2"
              className="object-cover max-h-[60vh] portrait:max-w-[70vw] rounded-lg"
              src={breakTimeNotification.images[currentSlide]}
              title="Pausenhinweis"
              alt=""
            />
            {/* Button right*/}
            <div className="flex justify-end items-center w-12">
              {currentSlide < breakTimeNotification.images.length - 1 && (
                <StyledButton
                  shape="smallSquare"
                  className="bg-whitetrans rounded-xl w-full  "
                  onClick={() => {
                    setCurrentSlide(currentSlide + 1);
                  }}
                >
                  <p className="">&#10140;</p>
                </StyledButton>
              )}
            </div>
          </div>
        </div>

        {/* Image Slide Navigation Dots */}
        {breakTimeNotification.images.length > 1 && (
          <div className="z-10 flex gap-4 lg:gap-6 slider-nav">
            {breakTimeNotification.images.map((_, index) => {
              return (
                <button
                  className={
                    (currentSlide === index ? "bg-yellow-400 " : "bg-white ") +
                    "w-2 h-2 transition duration-200 ease-in-out rounded-full opacity-75 lg:w-4 lg:h-4 hover:opacity-100"
                  }
                  data-testid={"breakSliderButton" + index}
                  onClick={() => setCurrentSlide(index)}
                />
              );
            })}
          </div>
        )}
      </div>
    </div>
  );
}
