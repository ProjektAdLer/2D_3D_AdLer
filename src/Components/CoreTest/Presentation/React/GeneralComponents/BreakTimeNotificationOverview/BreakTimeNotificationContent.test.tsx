import React from "react";
import BreakTimeNotificationContent from "../../../../../Core/Presentation/React/GeneralComponents/BreakTimeNotificationOverview/BreakTimeNotificationContent";
import { act, fireEvent, render } from "@testing-library/react";
import { shortBreakTimeNotificationContents } from "../../../../../Core/Domain/BreakTimeNotifications/BreakTimeNotifications";

describe("BreakTimeNotificationContent", () => {
  test("should render", () => {
    const renderResult = render(
      <BreakTimeNotificationContent
        breakTimeNotification={shortBreakTimeNotificationContents[0]}
      />,
    );

    expect(renderResult.container).not.toBeEmptyDOMElement();
    expect(renderResult.container).toMatchSnapshot();
  });

  // ANF-ID: [EKJ0003]
  test("click on navigation dot changes current slide image", () => {
    const renderResult = render(
      <BreakTimeNotificationContent
        breakTimeNotification={shortBreakTimeNotificationContents[0]}
      />,
    );

    act(() => {
      const navigationDots = renderResult.getAllByRole("button");
      fireEvent.click(navigationDots[1]);
    });

    expect(renderResult.getByRole("presentation").getAttribute("src")).toBe(
      shortBreakTimeNotificationContents[0].images[1],
    );
  });
});
