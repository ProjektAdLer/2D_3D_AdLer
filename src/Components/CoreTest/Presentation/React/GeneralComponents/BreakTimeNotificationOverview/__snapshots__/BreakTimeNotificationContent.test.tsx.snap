// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreakTimeNotificationContent should render 1`] = `
<div>
  <div
    class="pb-4 max-h-[90vh] portrait:max-w-[90vw]"
  >
    <div
      class="flex flex-col items-center gap-6 mx-0 my-auto slider-wrapper"
    >
      <div
        class="flex flex-col gap-4 overflow-y-auto items-center justify-center slider portrait:max-w-[90vw] max-h-[90vh]"
      >
        <p
          class="w-full pb-4 pl-6 text-sm font-bold md:text-xl lg:text-xl max-w-[60vw] ultraWide:max-w-[40vw] text-adlerdarkblue"
        >
          shortBreakBase 
          infoMobilization 
        </p>
        <div
          class="w-full flex flex-row justify-center align-center"
        >
          <div
            class="flex justify-start items-center w-12"
          />
          <img
            alt=""
            class="object-cover max-h-[60vh] portrait:max-w-[70vw] rounded-lg"
            id="slide-2"
            src="smallBreakContent-B3-slide1.png"
            title="Pausenhinweis"
          />
          <div
            class="flex justify-end items-center w-12"
          >
            <div
              class=""
            >
              <button
                class="w-full justify-center px-2 py-1 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
              >
                <p
                  class=""
                >
                  âžœ
                </p>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="z-10 flex gap-4 lg:gap-6 slider-nav"
      >
        <button
          class="bg-yellow-400 w-2 h-2 transition duration-200 ease-in-out rounded-full opacity-75 lg:w-4 lg:h-4 hover:opacity-100"
          data-testid="breakSliderButton0"
        />
        <button
          class="bg-white w-2 h-2 transition duration-200 ease-in-out rounded-full opacity-75 lg:w-4 lg:h-4 hover:opacity-100"
          data-testid="breakSliderButton1"
        />
      </div>
    </div>
  </div>
</div>
`;
